R/generate_hub_baselines.R

```diff
@@ -0,0 +1,237 @@
+    dplyr::filter(.data$target == !!target_name) |>
+    dplyr::mutate(
+      geo_value = forecasttools::us_location_recode(
+        .data$location,
+        "code",
+        "abbr"
+      )
+    ) |>
+    dplyr::rename(
+      time_value = "date"
+    ) |>
+    dplyr::select(-c("location", "target")) |>
+    epiprocess::as_epi_df()
+
+  check_data_latency(
+    epi_df,
+    reference_date,
+    desired_max_time_value,
+    target_label
+  )
+
+  args_list <- epipredict::cdc_baseline_args_list(aheads = 1:4, nsims = 1e5)
+  preds <- withr::with_rng_version(
+    "4.0.0",
+    withr::with_seed(rng_seed, {
+      fcst <- epipredict::cdc_baseline_forecaster(
+        epi_df |>
+          dplyr::filter(
+            .data$time_value <= desired_max_time_value
+          ),
+        "observation",
+        args_list
+      )
+      # advance forecast_date by a week due to data latency and
+      # create forecast for horizon -1
+      fcst$predictions |>
+        dplyr::mutate(
+          forecast_date = reference_date,
+          ahead = as.integer(.data$target_date - reference_date) %/% 7L
+        ) |>
+        # prepare -1 horizon predictions
+        dplyr::bind_rows(
+          epi_df |>
+            tidyr::drop_na(.data$observation) |>
+            dplyr::slice_max(.data$time_value) |>
+            dplyr::transmute(
+              forecast_date = reference_date,
+              target_date = reference_date - 7L,
+              ahead = -1L,
+              geo_value,
+              .pred = .data$observation,
+              .pred_distn = hardhat::quantile_pred(
+                values = matrix(
+                  rep(
+                    .data$observation,
+                    each = length(args_list$quantile_levels)
+                  ),
+                  nrow = length(.data$observation),
+                  ncol = length(args_list$quantile_levels),
+                  byrow = TRUE
+                ),
+                quantile_levels = args_list$quantile_levels
+              )
+            )
+        )
+    })
+  )
+
+  preds_formatted <- preds |>
+    epipredict::flusight_hub_formatter(
+      target = target_name,
+      output_type = "quantile"
+    ) |>
+    tidyr::drop_na(.data$output_type_id) |>
+    dplyr::arrange(.data$target, .data$horizon, .data$location) |>
+    dplyr::select(
+      "reference_date",
+      "horizon",
+      "target",
+      "target_end_date",
+      "location",
+      "output_type",
+      "output_type_id",
+      "value"
+    )
+  return(preds_formatted)
+}
+
+
+#' Generate hub baseline forecasts for a given disease and reference date
+#'
+#' @param base_hub_path Path to the base hub directory.
+#' @param reference_date Reference date (should be a Saturday).
+#' @param disease Disease name ("covid" or "rsv").
+#' @export
+generate_hub_baseline <- function(
+  base_hub_path,
+  reference_date,
+  disease
+) {
+  if (!disease %in% c("covid", "rsv")) {
+    stop("'disease' must be either 'covid' or 'rsv'")
+  }
```

Comment:
Can use `checkmate` here
```suggestion
   checkmate::assert_scalar(disease)
   checkmate::assert_names(disease, subset.of = c("covid", "rsv"))
```